core/matrix/dense.cpp

```diff
@@ -266,8 +270,20 @@ void Dense<ValueType>::inv_scale_impl(const LinOp *alpha)
         GKO_ASSERT_EQUAL_COLS(this, alpha);
     }
     auto exec = this->get_executor();
-    exec->run(dense::make_inv_scale(
-        make_temporary_conversion<ValueType>(alpha).get(), this));
+    // if alpha is real (convertible to real) and ValueType complex
+    if (dynamic_cast<const ConvertibleTo<Dense<>> *>(alpha) &&
```

Comment:
Is checking converting to Dense<double> enough?

Reply:
Yes, as all Dense formats are inter-convertible (double->float->double), this can be used to check whether something is a real dense vector.